sdm/src/main/java/com/sap/cds/sdm/persistence/DBQuery.java

```diff
@@ -29,6 +29,17 @@ public static Result getAttachmentsForUPID(
     return persistenceService.run(q);
   }
 
+  public static String getAttachmentForID(
+      CdsEntity attachmentEntity, PersistenceService persistenceService, String id) {
+    CqnSelect q =
+        Select.from(attachmentEntity).columns("fileName").where(doc -> doc.get("ID").eq(id));
+    Result result = persistenceService.run(q);
+    if (result.rowCount() == 0) {
+      return null;
+    }
+    return result.list().get(0).get("fileName").toString();
```

Comment:
add like this return result.rowCount() ==0? null:result.list().get(0).get("fileName").toString();

Reply:
Done